### ISSUE TITLE
The Hessian of the `LinearAlgebra.jl` dot is incorrect

### ISSUE THREAD

**Issue by Nimrais** [JuliaDiff/ForwardDiff.jl#622]
The computation of the Hessian of the `LinearAlgebra.jl` dot is wrong:

```
using ForwardDiff
using LinearAlgebra
@info ForwardDiff.hessian(x -> dot(x, ones(1,1), x), [0])
```
Produced output:
```
[ Info: [0.0;;]
```

Environment:
```
julia version 1.8.5
```

**Comment by KristofferC** [comment]
Fixed by https://github.com/JuliaDiff/ForwardDiff.jl/pull/481

**Comment by mcabbott** [comment]
Fixed by #481, which is presently on master but not the tagged version:
```
julia> ForwardDiff.hessian(x -> dot(x, ones(1,1), x), [0])
1×1 Matrix{Float64}:
 2.0

(jl_zBhOAL) pkg> st ForwardDiff
Status `/private/var/folders/yq/4p2zwd614y59gszh7y9ypyhh0000gn/T/jl_zBhOAL/Project.toml`
  [f6369f11] ForwardDiff v0.11.0-DEV `~/.julia/dev/ForwardDiff`
```
